go-controller/pkg/ovn/zone_interconnect/interconnect.go

```diff
@@ -0,0 +1,524 @@
+	}
+
+	lspOptions := map[string]string{
+		"router-port":       logicalRouterPortName,
+		"requested-tnl-key": strconv.Itoa(nodeID),
+	}
+
+	err = zic.addNodeLogicalSwitchPort(zic.networkTransitSwitchName, zic.getNetworkScopedName(types.TransitSwitchToRouterPrefix+node.Name),
+		"router", []string{"router"}, lspOptions)
+	if err != nil {
+		return err
+	}
+
+	// Its possible that node is moved from a remote zone to the local zone. Check and delete the remote zone routes
+	// for this node as it's no longer needed.
+	return zic.deleteLocalNodeStaticRoutes(node, nodeID, nodeTransitSwitchPortIPs)
+}
+
+func (zic *ZoneInterconnectHandler) createRemoteZoneNodeResources(node *corev1.Node, nodeID int, chassisId string) error {
+	nodeTransitSwitchPortIPs, err := util.ParseNodeTransitSwitchPortAddrs(node)
+	if err != nil || len(nodeTransitSwitchPortIPs) == 0 {
+		return fmt.Errorf("failed to get the node transit switch port Ips : %v", err)
+	}
+
+	ts := &nbdb.LogicalSwitch{
+		Name: zic.networkTransitSwitchName,
+	}
+
+	_, err = libovsdbops.GetLogicalSwitch(zic.nbClient, ts)
+	if err != nil {
+		// Create transit switch
+		ts.OtherConfig = map[string]string{
+			"interconn-ts":             zic.networkTransitSwitchName,
+			"requested-tnl-key":        transitSwitchTunnelKey,
+			"mcast_snoop":              "true",
+			"mcast_flood_unregistered": "true",
+		}
+		if err := libovsdbops.CreateOrUpdateLogicalSwitch(zic.nbClient, ts); err != nil {
+			return fmt.Errorf("failed to create/update logical transit switch: %v", err)
+		}
+	}
+
+	var transitRouterPortMac net.HardwareAddr
+	var transitRouterPortNetworks []string
+	foundIPv4 := false
+	for _, ip := range nodeTransitSwitchPortIPs {
+		if utilnet.IsIPv4CIDR(ip) {
+			foundIPv4 = true
+			transitRouterPortMac = util.IPAddrToHWAddr(ip.IP)
+		} else {
+			if !foundIPv4 {
+				transitRouterPortMac = util.IPAddrToHWAddr(ip.IP)
+			}
+		}
+		transitRouterPortNetworks = append(transitRouterPortNetworks, ip.String())
+	}
+
+	remotePortAddr := transitRouterPortMac.String()
+	for _, tsNetwork := range transitRouterPortNetworks {
+		remotePortAddr = remotePortAddr + " " + tsNetwork
+	}
+
+	lspOptions := map[string]string{
+		"requested-tnl-key": strconv.Itoa(nodeID),
+	}
+	remotePortName := zic.getNetworkScopedName(types.TransitSwitchToRouterPrefix + node.Name)
+	if err := zic.addNodeLogicalSwitchPort(zic.networkTransitSwitchName, remotePortName, "remote", []string{remotePortAddr}, lspOptions); err != nil {
+		return err
+	}
+	// Set the port binding chassis.
+	if err := zic.setRemotePortBindingChassis(node.Name, remotePortName, chassisId); err != nil {
+		return err
+	}
+
+	if err := zic.addRemoteNodeStaticRoutes(node, nodeID, nodeTransitSwitchPortIPs); err != nil {
+		return err
+	}
+
+	// Cleanup the logical router port connecting to the transit switch for the remote node (if present)
+	// Cleanup would be required when a local zone node moves to a remote zone.
```

Comment:
I see there is some cleanup done here as well I guess you can dis-regard my previous comment then.